packages/block-editor/src/components/block-icon/index.native.js

```diff
@@ -8,46 +8,40 @@ import { View } from 'react-native';
  */
 import { Icon } from '@wordpress/components';
 import { blockDefault } from '@wordpress/icons';
-import { withPreferredColorScheme } from '@wordpress/compose';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import styles from './style.scss';
 
-export function BlockIcon( {
-	icon,
-	fill,
-	size,
-	showColors = false,
-	getStylesFromColorScheme,
-} ) {
+export function BlockIcon( { icon, fill, size, showColors = false } ) {
 	if ( icon?.src === 'block-default' ) {
 		icon = {
 			src: blockDefault,
 		};
 	}
 
+	const defaultFill = usePreferredColorSchemeStyle(
+		styles.iconPlaceholder,
+		styles.iconPlaceholderDark
+	)?.fill;
+	const iconForeground = showColors ? icon?.foreground : undefined;
+
 	const renderedIcon = (
 		<Icon
 			icon={ icon && icon.src ? icon.src : icon }
-			{ ...( fill && { fill } ) }
+			fill={ fill || iconForeground || defaultFill }
 			{ ...( size && { size } ) }
-			{ ...( ! fill &&
-				getStylesFromColorScheme(
-					styles.iconPlaceholder,
-					styles.iconPlaceholderDark
-				) ) }
 		/>
 	);
 	const style = showColors
 		? {
 				backgroundColor: icon && icon.background,
-				color: icon && icon.foreground,
```

Comment:
`color` style property is not supported in the `View` component ([reference](https://reactnative.dev/docs/0.64/view-style-props)) so this line wasn't really applying the foreground color defined in the icon.

Reply:
Makes sense. Good eye here @fluiddot 👍🏾